### PR TITLE
Testing infrastructure for disclosed contracts

### PR DESCRIPTION
**Pull request by carlpulley-da**
CHANGELOG_BEGIN

* Changes to facilitate testing explicit disclosure under #14227.

CHANGELOG_END

- [x] expose disclosed contracts to `Speedy.Machine.from*` methods (with a default empty contract value)
- [x] expose disclosed contracts in `EvaluationOrderTest#evalUpdateApp` (with a default empty contract value)

Currently, no changes are proposed for [StoreBackedCommandExecutorSpec.scala](https://github.com/digital-asset/daml/blob/12a488f6a49c6676a94e126a61fe54a796714d68/ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/execution/StoreBackedCommandExecutorSpec.scala) (i.e. for engine level testing) since it looks like we can create mock engine instances and specify disclosed contracts on submission.

### CODE REVIEW

File: daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/Runner.scala [hunk 1]
```diff
@@ -428,9 +428,12 @@ private[lf] class Runner(
...
     val machine =
-      Speedy.Machine.fromPureSExpr(extendedCompiledPackages, script.expr, traceLog, warningLog)(
-        Script.DummyLoggingContext
-      )
+      Speedy.Machine.fromPureSExpr(
+        extendedCompiledPackages,
+        script.expr,
+        traceLog = traceLog,
+        warningLog = warningLog,
+      )(Script.DummyLoggingContext)
```
Comment (carlpulley-da) [inline daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/Runner.scala:436]: As disclosed contracts is (potentially) exposed here, we now need default parameters to be named.

File: daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala [hunk 2]
```diff
@@ -117,7 +117,7 @@ class Engine(val config: EngineConfig = Engine.StableConfig) {
     for {
```
Comment (carlpulley-da) [inline daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala:117]: All changes in this file should be boy scouting work

File: daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Transaction.scala [hunk 3]
```diff
@@ -701,6 +701,7 @@ object Transaction {
     * @param nodeSeeds        : An association list that maps to each ID of create and exercise
     *                         nodes its seeds.
     * @param globalKeyMapping : input key mapping inferred by interpretation
+    * @param disclosures      : contracts explicitly disclosed to this transaction
```
Comment (carlpulley-da) [inline daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Transaction.scala:704]: Boy scouting work

File: security-evidence.md [hunk 4]
```diff
@@ -61,134 +61,134 @@
 - /v1/query POST succeeds after reconnect to DB: [FailureTests.scala](ledger-service/http-json/src/failurelib/scala/http/FailureTests.scala#L260)
```
Comment (carlpulley-da) [inline security-evidence.md:61]: All changes in this file are auto generated from changes to EvaluationOrderTest?

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala [hunk 5]
```diff
@@ -111,7 +111,7 @@ private[lf] object Speedy {
     private[speedy] val any = SValue.SAny(TTyCon(templateId), value)
   }
 
-  private[this] def enforceLimit(actual: Int, limit: Int, error: Int => IError.Limit.Error) =
+  private[this] def enforceLimit(actual: Int, limit: Int, error: Int => IError.Limit.Error): Unit =
```
[the remark this answers is not in the document]
Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala:114]: No problem. 

In general, I do not impose the return type for private method.  I let you judge if we need it here or not.

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala [hunk 6]
```diff
@@ -216,7 +216,7 @@ private[lf] object Speedy {
   )
 
   object DisclosureTable {
-    val Empty = DisclosureTable(Map.empty, Map.empty)
+    val Empty: DisclosureTable = DisclosureTable(Map.empty, Map.empty)
```
Comment (remyhaemmerle-da) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala:219]: This is not private, so I prefer we put the type. :+1:

**Review by remyhaemmerle-da — APPROVED** [review]
LGTM. 

I really like the boy scout work you have done.

**Review by cocreature — APPROVED** [review]
Thank you!

**Pull request merged**